### PR TITLE
Add support for path stats in cli and web api

### PR DESCRIPTION
**Pull request by klingerf**
The cli stat command supports grouping by pod and deployment. With this change, it will also support grouping by path, in order to facilitate summary stats per individual endpoint.

### CODE REVIEW

File: controller/api/public/grpc_server.go [hunk 1]
```diff
@@ -467,6 +474,10 @@ func processRequestRate(
 
 		datapoints := make([]*pb.MetricDatapoint, 0)
 		for _, value := range metric.Values {
+			if value.Value == 0 {
+				continue
+			}
```
Comment (klingerf) [inline controller/api/public/grpc_server.go]: This change makes the request count metric behave the same as the success rate metric. Specifically, we will no longer return a datapoint if request count == 0; it's already the case that we don't return a success rate datapoint when request count == 0. We should address how to handle sparse timeseries in a subsequent pull request.

File: cli/cmd/stat.go [hunk 2]
```diff
@@ -36,6 +36,7 @@ var statCmd = &cobra.Command{
 Valid resource types include:
  * pods (aka pod, po)
  * deployments (aka deployment, deploy)
+ * paths (aka path, pa)
```
Comment (rmars) [inline cli/cmd/stat.go]: small nit: we should add paths to the error messagea on line 51

File: cli/cmd/stat.go [hunk 3]
```diff
@@ -142,7 +147,7 @@ func displayStats(resp *pb.MetricResponse, w *tabwriter.Writer) {
 	for _, name := range sortedNames {
 		fmt.Fprintf(
 			w,
-			"%s\t%frps\t%f%%\t%dms\t%dms\n",
+			"%s\t%.2frps\t%.2f%%\t%dms\t%dms\n",
```
Comment (olix0r) [inline cli/cmd/stat.go]: I think the request rate only needs a single digit of precision -- if we only get one request in a window, the smallest value we can represent is `0.1`

Also, I'd really really like these to be right-aligned (i.e. in the column).i'd be happy to address this in a followup if you'd prefer, though.

Comment (klingerf) [reply cli/cmd/stat.go]: Cool, all makes sense. I took a stab at the alignment in 84ce6959. Looks like this:

```
NAME            REQUEST_RATE   SUCCESS_RATE   P50_LATENCY   P99_LATENCY
default/hello         7.5rps        100.00%           0ms          32ms
default/world         7.7rps        100.00%           0ms          12ms
```

**Pull request merged**